Extensions/xkit_patches.js

```diff
@@ -331,44 +389,73 @@ XKit.tools.getParameterByName = function(name){
 
 		XKit.iframe = {
 
+			/**
+			 * @return {String} Id of blog which the iframe refers to (usually
+			 *                  the blog in which the iframe is embedded)
+			 */
 			get_tumblelog: function() {
 				var new_channel_id = document.location.href.match(/[&?]tumblelogName=([\w-]+)/);
 				var old_channel_id = $("#tumblelog_name").attr('data-tumblelog-name');
 				return (new_channel_id && new_channel_id[1]) || old_channel_id;
 			},
 
+			/**
+			 * @return {String?} Post which this iframe refers to or undefined
+			 */
 			single_post_id: function() {
 				var all_post_ids = document.location.href.match(/[&?](singlePostId|pid|postId)=(\d+)/);
 				return all_post_ids[2];
 			},
 
+			/**
+			 * @return {String} Form key of the iframe (the data to use in a
+			 *                  reblog or other API request)
+			 */
 			form_key: function() {
 				var new_form_key = $("meta[name=tumblr-form-key]").attr("content");
 				var old_form_key = $(".btn.reblog").attr('data-form-key');
 		        return new_form_key || old_form_key;
 			},
 
+			/**
+			 * @param {jQueryElement} button - The button in the iframe to be hidden
+			 */
 			hide_button: function(button) {
 				button.addClass("no_label");
 				button.addClass("no-text").wrapInner('<span class="hidden">');
 			},
 
+			/**
+			 * @return {jQueryElement} The follow button in the iframe
```

Comment:
I've been using jQueryElement as the return type of `$(anything)`. jQuerySelector might be slightly clearer. TypeScript uses `JQuery` as their return type, which I think I'll use for consistency unless you have a different preference.